tests/test_spec_set_functions.py

```diff
@@ -14,3 +19,190 @@ def test_existence():
     assert ragged.unique_counts is not None
     assert ragged.unique_inverse is not None
     assert ragged.unique_values is not None
+
+
+# unique_values tests
+def test_can_take_none():
+    with pytest.raises(TypeError):
+        assert ragged.unique_values(ragged.array(None)) is None
+
+
+def test_can_take_list():
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "the truth value of an array whose length is not 1 is ambiguous;"
+        ),
+    ):
+        assert ragged.unique_values(
+            ragged.array([1, 2, 4, 3, 4, 5, 6, 20])
+        ) == ragged.array([1, 2, 3, 4, 5, 6, 20])
+
+
+def test_can_take_empty_arr():
+    with pytest.raises(TypeError):
+        assert ragged.unique_values(ragged.array([])) == ragged.array([])
+
+
+def test_can_take_moredimensions():
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "the truth value of an array whose length is not 1 is ambiguous;"
+        ),
```

Comment:
This is not the error message that users should see if they pass a multidimensional array to `ragged.unique_values` and it isn't supported. It should be a `NotImplementedError` with an error message saying that multidimensional arrays aren't supported, but flat or rectilinear (i.e. NumPy-like) arrays are. That is, the error message should tell them that they didn't call the function the wrong way, but what they're trying to do hasn't been implemented yet and what they can do instead. This error message is due to an implementation detail (it will change if the implementation changes a little bit) and it doesn't tell a user—who ought to know how to use the library but isn't expected to know how it's implemented—what they need to know.